### DISCUSSION TITLE
How to update External Domain?

### DISCUSSION THREAD

**Discussion by ChristianLavigne** [zitadel/zitadel#8351] [Q&A] [answered]
Hi,
I have a self-hosted instance (docker) that works pretty well. But I need to change the External Domain.

I have updated the environment variable ZITADEL_EXTERNALDOMAIN to the appropriate domain, but the changes are not applied.

I read in the documentation that I need to run the setup phase, but I can't find how to do this anywhere?! And I do not want to lose my configuration.

Q: How can I update External Domain?

**Comment by eliobischof** [comment]
If your container is run with the `zitadel start-from-init` or `zitadel start-from-setup` command, changing ZITADEL_EXTERNALDOMAIN and then running  `docker compose up` should be enough

**Reply by ChristianLavigne** [reply]
Thanks for the response. 

I tried that already, I also tried docker-compose down before too. My container run parameters are:

- command: 'start-from-init --masterkeyFromEnv --tlsMode external'

But it did not replace the original domain everywhere. For example: I still get the old domain on the link in the validation emails.

**Reply by HungryHowies** [reply]
Hey @ChristianLavigne 

 I'm not sure if this helps and I don't use docker , but for changes in my  database I run the following. 

```
zitadel setup  --init-projections=true   --config defaults.yaml  --masterkey "MasterkeyNeedsToHave32Characters"  --tlsMode external
```

**Reply by ChristianLavigne** [reply]
Thanks @HungryHowies 

That is a good lead! Unfortunately, --init-projections doesn't work with docker, and I could not find the equivalent.

And I think the issue go deeper than the projections though. I still see the domain I'm trying to get rid of (original) in the instance domains, but I cannot find a way to remove it. New domain is set as the default and the original is not in the settings | domain.

Here is what I see in the console (instance level):
![Screenshot from 2024-07-24 07-50-18](https://github.com/user-attachments/assets/c5006ea4-f830-4160-ba18-7fbf2eea8701)

I guess I'm going to have to rebuild this thing and be more judicious with my choice of name. Fortunately, I'm not too far along and it is still possible. 

This gives me pause on deploying this to production. What if I need to change some settings in the future, and that I can't...

**Reply by HungryHowies** [reply, marked as answer]
I understand, What we did was  build from source, this may not be  a choice you would want but for us it gave us a lot of options.

**Reply by ChristianLavigne** [reply]
@HungryHowies 
Building from source is a great solution, we're just not willing to invest that much at this point.

We decided we will be rebuilding the instance.

Thanks for the info,
Cheers!
